File: frontend/src/app/shared/service/admin/common/ClassAverageBonusAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

// import {environment} from '../../../../../environments/environment';
import {ClassAverageBonusDto} from '../../../model/common/ClassAverageBonus.model';

import {ClassAverageBonusCriteria} from '../../../criteria/common/ClassAverageBonusCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";
import {environment} from "../../../../../environments/environment";


@Injectable({
    providedIn: 'root'
})
export class ClassAverageBonusAdminService extends AbstractService<ClassAverageBonusDto, ClassAverageBonusCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'admin/classAverageBonus/';
    }

    public constrcutDto(): ClassAverageBonusDto {
        return new ClassAverageBonusDto();
    }

    public constrcutCriteria(): ClassAverageBonusCriteria {
        return new ClassAverageBonusCriteria();
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/admin/common/ClassAverageBonusAdminService.java
package ma.zs.alc.service.facade.admin.common;

import java.util.List;
import ma.zs.alc.bean.core.common.ClassAverageBonus;
import ma.zs.alc.dao.criteria.core.common.ClassAverageBonusCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface ClassAverageBonusAdminService extends  IService<ClassAverageBonus,ClassAverageBonusCriteria>  {

    List<ClassAverageBonus> findByCollaboratorId(Long id);
    int deleteByCollaboratorId(Long id);
    long countByCollaboratorId(Long id);




}
